Automattic/wp-calypso
# Pattern Library: Share more code between SSR and client-side rendering

Related to #87996
Fixes https://github.com/Automattic/dotcom-forge/issues/5867

## Proposed Changes

After a discussion (p1709290737687439-slack-C06ELHR6L9J) with @nightnei in Slack, I realized that we can actually share more code between server-side rendering and client-side rendering. We just need to be careful about not importing any of the pattern preview components from Assembler in any of the SSR components.

This PR makes `Patterns` accept a `patternGallery` prop, which supplies a component to be used for rendering a gallery of patterns. This way, we can use the same component for the entire page structure and just swap out the pattern gallery piece as needed.

## Testing Instructions



1. Sandbox `public-api.wordpress.com`
2. Apply this back-end patch 33ae5-pb/#diff
3. Go to `/patterns` in an incognito window
4. Ensure that the page renders as expected

## Pre-merge Checklist
